frontend-zs/src/components/PrivateRoute.js

```diff
@@ -3,11 +3,13 @@ import { Route, Redirect } from 'react-router-dom';
 
 import { authenticationService } from '../services/authentication.service';
 
-export const PrivateRoute = ({ component: Component }) => (
-  <Route render={props => {
-    const currentUser = authenticationService.currentUserValue;
-    if (!currentUser) return <Redirect to={{ pathname: '/login' }} />
-    if (!currentUser.registrationComplete) return <Redirect to={{ pathname: '/registrationStepTwo' }} />
-    return <Component/>
-  }} />
+export const PrivateRoute = ({ component: Component, ...rest }) => (
+  <Route
+    {...rest}
```

Comment:
what does this exactly fix?

Reply:
It keeps the id that is send with the path. For example path='/sotry/:id'. Otherwise the id is not identified in StoryPage.

Reply:
we only need to reorder the routes in `App.js` and it will be fixed. Use the route `/story/:id` first, then `/story`